packages/core-tabs/core-tabs.js

```diff
@@ -43,47 +45,110 @@ export default class CoreTabs extends HTMLElement {
+  get panels () { return this.tabs.map(getPanelFromTab) }
 
-  get panel () { return this.panels.filter((panel) => panel && !panel.hasAttribute('hidden'))[0] }
+  get panel () { return getPanelFromTab(this.tab) }
 
+  /**
+   * @returns {[TabElement]}
+   */
   get tabs () { return queryAll('button,a', this) }
 
-  get tab () { return document.getElementById(this.panel && this.panel.getAttribute(FROM)) }
+  /**
+   * @returns {TabElement}
+   */
+  get tab () {
+    const tabAttr = this.getAttribute('tab')
+    const allTabs = this.tabs
+    const allPanels = this.panels
+
+    let tab = isInteger(parseInt(tabAttr))
+      ? allTabs[tabAttr] // Integer, assume index
+      : document.getElementById(tabAttr) // Non-integer, assume id
+
+    if (!tab) { // No tab is set, check for hidden panels
+      const firstVisiblePanel = allPanels.filter(panel => panel && !panel.hasAttribute('hidden'))[0]
+      tab = allTabs.filter(tab => tab.getAttribute('aria-controls') === (firstVisiblePanel && firstVisiblePanel.id))[0]
+    }
+
+    if (!tab) { tab = allTabs[0] } // No tab, fallback to first tab
+    return tab
+  }
 
+  /**
+   * @param {string | number | TabElement} value
+   * @returns {void}
+   */
   set tab (value) {
     if (!value && value !== 0) return
-    const panels = this.panels
-    const prevIndex = this.tabs.indexOf(this.tab)
-    const nextIndex = this.tabs.reduce((acc, tab, i) => {
-      return (i === value || tab === value || tab.id === value) ? i : acc
-    }, this.tab)
-
-    this.tabs.forEach((tab, index) => {
-      const panel = panels[index]
-      const openTab = index === nextIndex
-      const openPanel = panel === panels[nextIndex]
+    const allTabs = this.tabs
+    const allPanels = this.panels
+    const prevIndex = allTabs.indexOf(this.tab)
+    const nextIndex = allTabs.indexOf(allTabs.filter((tab, i) => (i === Number(value) || tab === value || tab.id === value))[0] || this.tab)
+
+    /** @type {TabElement | undefined} */
+    let nextTab
 
-      tab.setAttribute('aria-selected', openTab)
-      tab.setAttribute('tabindex', Number(openTab) - 1)
-      toggleAttribute(panel, 'hidden', !openPanel)
+    allTabs.forEach((tab, index) => {
+      const tabPanel = getPanelFromTab(tab)
+      const openTab = index === nextIndex
+      const openPanel = tabPanel === allPanels[nextIndex]
+
+      if (openTab && openPanel) nextTab = tab
+      tab.setAttribute('aria-selected', openTab && openPanel)
+      tab.setAttribute('tabindex', Number(openTab && openPanel) - 1)
+      if (tabPanel) {
+        toggleAttribute(tabPanel, 'hidden', !openPanel)
+        if (openTab) {
+          tabPanel.setAttribute(FROM, tab.id)
+        }
+      }
     })
 
-    if (prevIndex !== nextIndex) dispatchEvent(this, 'tabs.toggle')
+    if (prevIndex !== nextIndex) {
+      // Update tab-attribute if present.
+      const tabAttr = this.getAttribute('tab')
```

Comment:
Vi burde nok alltid oppdatere attributten tab, uansett om det er change eller ei (browseren skipper om det er samme verdi), men skal vi da force attributten til å bli `id`, eller kan den være av og til det ene og av og til det andre? synes attributten burde være én ting egentlig.. men vanskelig å velge mellom id og index - pros og cons på begge

Reply:
Enig i oppdatering uansett.
Testet litt med oppdatering som endrer til kun id; blir rart om man kan sette opp med index og så oppdateres den til en id. Lite gunstig ved f.eks debugging
Er det et alternativ å støtte begge deler? kan gjøre det lettere å bruke om man kan hente både `tabId` og `tabIndex`?
@ingvilddev har du noen tanker her?

Reply:
tenkt prioritering for å sette aktiv tab; `tabId` > `tabIndex` > `hidden` i panels > index `0`

Reply:
Vet ikke helt om jeg henger med, men jeg har forstått det som at `tab` er prop på jsx-komponent for å sette "current tab" --> `aria-selected="true"` på korrekt tab?

Men det ser ikke ut til å funke fra start: ⬇️ 
![image](https://user-images.githubusercontent.com/72200992/140712901-3b131ccf-fcb7-4b97-b247-5654468b1a92.png)
Første element er `aria-controls="false"`, selvom jeg ser når jeg debugger koden vår at vi sender inn `tab="tv-series-season-tabs-item-0"` (tror jeg).

For min del virker det greit å kun akseptere en `tabId`, og heller at core-components selv fallbacker til index om tab ikke er sendt inn. Ieg slet også med å forstå hva `tab` egentlig var, derav synes jeg der er bra at vi har fått på plass doc på det i https://github.com/nrkno/core-components/pull/636/files/279c245ef1dbfece2bdf02dcebbaef07a14037a4#diff-4916005a34d9f6872087b8a2db3edf5d725a4a670606a032dfa8086507f672d6R214, men jeg lurte originalt på om en mer forklarende navn hadde hjulpet, siden verdien ikke er 1-1 til et standard html-attributt, `currentTabId` f.eks., men jeg er jo ikke _i_ i materien som dere er, og jeg skjønne at en navneendring vil brekke eksisterende kode, men ville bare lufte tankene jeg hadde som konsument (sry @eirikbacker , lots of text and thoughts again 😞 )

Reply:
Hehe, dette er nok en god plan at vi to snakker om muntlig så kommer vi frem til noe smart er jeg sikker på @skjalgepalg  <3